Content.Server/Radio/Components/RadioJammerComponent.cs

```diff
@@ -9,12 +9,25 @@ namespace Content.Server.Radio.Components;
 [Access(typeof(JammerSystem))]
 public sealed partial class RadioJammerComponent : Component
 {
-    [DataField("range"), ViewVariables(VVAccess.ReadWrite)]
-    public float Range = 8f;
+    [DataDefinition]
+    public partial struct RadioJamSetting
+    {
+        [DataField(required: true)]
+        public float Wattage;
 
-    /// <summary>
-    /// Power usage per second when enabled
-    /// </summary>
-    [DataField("wattage"), ViewVariables(VVAccess.ReadWrite)]
-    public float Wattage = 2f;
+        [DataField(required: true)]
+        public float Range;
+
+        [DataField(required: true)]
+        public LocId Message = string.Empty;
+
+        [DataField(required: true)]
+        public LocId Name = string.Empty;
+    }
+
+    [DataField(required: true)]
+    public List<RadioJamSetting> Settings = new();
+
+    [DataField]
+    public byte SelectedPowerLevel = 1;
```

Comment:
this pattern is fine but make it in an `int` instead of a byte.

Reply:
> this pattern is fine but make it in an `int` instead of a byte.

why int instead of byte?

Reply:
I also have the same question! I only used a byte because it seemed like most enums used bytes and this is *kind of* a similar thing. Is it just because its cleaner to use ints for everything except enums to avoid confusion / issues?

Just pushed an update for this!
I'm going to just push the updated sprite, finals week is coming up so I can't add the battery level stuff but the updated sprite looks a lot better (At least to me :laughing:).

Reply:
> I also have the same question! I only used a byte because it seemed like most enums used bytes and this is _kind of_ a similar thing. Is it just because its cleaner to use ints for everything except enums to avoid confusion / issues?
> 
> Just pushed an update for this! I'm going to just push the updated sprite, finals week is coming up so I can't add the battery level stuff but the updated sprite looks a lot better (At least to me 😆).
> 
> (Also I don't know why the tests are failing... Hopefully its githubs fault not mine!)

It seems to be your fault?
> Resources/Textures/Objects/Devices/jammer.rsi: meta.json: [$.license] 'CC0 1.0 Universal' is not one of ['CC-BY-3.0', 'CC-BY-4.0', 'CC-BY-SA-3.0', 'CC-BY-SA-4.0', 'CC-BY-NC-3.0', 'CC-BY-NC-4.0', 'CC-BY-NC-SA-3.0', 'CC-BY-NC-SA-4.0', 'CC0-1.0']

Reply:
Fixed! Thanks.